dbus/jackdbus.c

```diff
@@ -709,17 +709,21 @@ paths_init()
 	    if (!(xdg_config_home = pathname_cat(home_dir, DEFAULT_XDG_CONFIG))) goto fail;
 	}
 
-    if (!(xdg_log_home = pathname_cat(home_dir, DEFAULT_XDG_LOG))) goto fail;
+    xdg_state_home = getenv("XDG_STATE_HOME");
+    if (xdg_state_home == NULL)
+    {
+        if (!(xdg_state_home = pathname_cat(home_dir, DEFAULT_XDG_STATE))) goto fail;
```

Comment:
This leaks memory and is problematic in general. A pointer from `getenv()` shouldn't be `free()` d, but `pathname_cat()` returns an allocated buffer pointer which definitely should. If you use a single pointer variable like this, you lose information about whether you need to call `free()` on it before returning.

That being said, none of the temporary allocations from `pathname_cat()` are ever freed. The whole thing seems to have been leaky since 2008.